bin/webpack-dev-server.js

```diff
@@ -331,6 +331,8 @@ function startDevServer(config, options) {
 
   let compiler;
 
+  config.output = Object.assign({}, config.output, { publicPath: createDomain(options) + options.publicPath });
```

Comment:
We need to check for an existing `config.output.publicPath` e.g

```js
const publicPath = options.publicPath || config.output.PublicPath

config.output = Object.assign({}, config.output, { publicPath: createDomain(options) + publicPath });
```

Reply:
Yeah the `public` option feels strange, and it would make sense to use `https`, `host`, `port` and `publicPath`, but that would be a breaking change.